File: CPP/Arrays/swapAlternate.cpp
#include <iostream>
using namespace std;


void swapAlternate(int arr[], int n){

    for (int i = 0; i < n; i = i+2){

        if(i+1<n){
            swap(arr[i],arr[i+1]);
        }
    }
}

void printArray(int arr[], int n){
    for (int i =0; i<n;i++){
        cout << arr[i] << " ";
    }
}

int main()
{

   int evenarray[6] = {1,2,3,4,5,6};
   swapAlternate(evenarray,6);

   int oddarray[5] = {210,5,534,95,6};
   swapAlternate(oddarray,5);

   cout << "The alternated array is : ";
   printArray(evenarray,6);

   cout << endl << endl;

   cout << "The alternated array is  : ";
   printArray(oddarray,5);



}

File: Java/Arrays/Swap_Alternate.java
/*
Problem Description: You are given with an array of length N, you have to swap every pair of
alternate elements in the array.
For example:N= 6
arr[] = 9 3 6 12 4 32
Output after swapping : 3 9 12 6 32 4

 */

import java.util.Scanner;

public class SwapAlternate {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.print("Enter Size of Array: ");
        int n = in.nextInt();

        // Input Array Elements
        int arr[] = new int[n];
        System.out.println("Enter " +n+ " Array Elements: ");
        for (int i = 0; i < n; i++) {
            arr[i] = in.nextInt();
        }

        // Logic to Swap Alternate Elements
        for (int i = 0; i < (arr.length - 1); i += 2) {
            int temp = arr[i];
            arr[i] = arr[i + 1];
            arr[i + 1] = temp;
        }

        // Print/Output updated new Array with alternate elements swapped
        for (int j = 0; j < n; j++) {
            System.out.print(arr[j]+ " ");
        }
    }
}
